### ISSUE TITLE
Add ability to map OAuth users to Grafana organizations

### ISSUE THREAD

**Issue by treydock** [grafana/grafana#23473]
**What would you like to be added**:

The ability for OIDC authentication to map people to a organization based on the claims returned during authentication.  This would be very similar to `role_attribute_path`.

Example:

```
org_attribute_path = contains(groups[*], 'staff') && 1 ||  3
```
**Why is this needed**:

Right now the only way to do anything close to this is use `auto_assign_org_id` then for people who need a different org, assign them to that org manually.  This doesn't actually work, see #22605.

This feature request may also resolve #16763

**Comment by seanlaff** [comment]
Huge proponent of this. We rely heavily on OAuth across our apis and apps. In grafana's case we use it to control our users' login and grafana permissions via `role_attribute_path`. Additionally, we make heavy use of the `Forward OAuth Identity` feature in the datasource settings. 

![Screen Shot 2020-06-01 at 1 48 25 PM](https://user-images.githubusercontent.com/3194832/83438188-faa77f80-a40e-11ea-8c6a-d82b58b60934.png)

Overall grafana provides a fantastic experience allowing our identity provider to pass this info downstream to grafana without relying on flaky, intermittent api calls for synchronization... however the dream falls short once orgs enter the picture since grafana doesn't support reading them from the token.

Any way we can help with this change?

**Comment by aknuds1** [comment]
Thanks, looking into this.

**Comment by treydock** [comment]
This will be resolved once #23661 is merged.

**Comment by aknuds1** [comment]
@treydock Thanks, I realized in the meantime.

**Comment by thomas315** [comment]
> This will be resolved once #23661 is merged.

I think this issue won't get resolved as mentioned in related closed issue.

**Comment by kminehart** [comment]
Since it's been a year since the last comment on this issue, I'm going to move this to a discussion since there's so many reactions in the pull request / this issue. I'll add more information in the discussion.
